src/tooltip.ts

```diff
@@ -138,14 +134,29 @@ class CompletionTooltip {
         if (this.info) this.view.requestMeasure(this.placeInfo)
       })
     }
-
     if (this.updateSelectedOption(open.selected)) {
       if (this.info) {this.info.remove(); this.info = null}
-      let option = open.options[open.selected]
-      if (option.completion.info) {
-        this.info = this.dom.appendChild(createInfoDialog(option, this.view)) as HTMLElement
+      let {completion} = open.options[open.selected]
+      let {info} = completion
+      if (!info) return
+      if (typeof info === 'string') {
+        this.info = this.dom.appendChild(createInfoDomNode(info))
         this.view.requestMeasure(this.placeInfo)
+        return;
+      }
+      let infoResult = info(completion)
+      if ('then' in infoResult) {
+        infoResult.then(maybeNode => {
+          if (!maybeNode) return
+          let newState = this.view.state.field(this.stateField)
```

Comment:
Thinking about doing this in case configuration changes while inflight

```suggestion
          let newState = this.view.state.field(this.stateField, false)
```

Reply:
Why do you need the new state at all here?

Reply:
To check if we're in the same context of completion. i.e. we're not slapping info of one completion on another

Reply:
Right, sorry, I figured that out as I was looking at the actual code.